### PR TITLE
[GraphBolt] Add multi-fanout per layer test.

### PR DESCRIPTION
**Pull request by yxy235**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by mfbalin — APPROVED** [review]
LGTM. I am assuming that if the CPU passes, it should be correct.

File: tests/python/pytorch/graphbolt/test_subgraph_sampler.py [hunk 1]
```diff
@@ -549,3 +549,44 @@ def test_SubgraphSampler_unique_csc_format_Hetero(labor):
                     sampled_subgraph.sampled_csc[etype].indptr,
                     csc_formats[step][etype].indptr.to(F.ctx()),
                 )
+
+
+@unittest.skipIf(
+    F._default_context_str == "gpu",
+    reason="Heterogenous sampling is not supported on GPU yet.",
+)
+@pytest.mark.parametrize("labor", [False, True])
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_subgraph_sampler.py:558]: should `TemporalNeighborSampler` be covered too?

**Review by Rhett-Ying — APPROVED** [review]

**Comment by mfbalin** [comment]
When this is merged, I will enable it in #6895.

File: tests/python/pytorch/graphbolt/test_subgraph_sampler.py [hunk 2]
```diff
@@ -872,3 +872,74 @@ def test_SubgraphSampler_unique_csc_format_Hetero(labor):
...
+                "n1:e1:n2": 4,
+                "n2:e2:n1": 2,
+            },
+            {
+                "n1:e1:n2": 2,
+                "n2:e2:n1": 1,
+            },
+        ]
+    for minibatch in sampler_dp:
+        print(minibatch)
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_subgraph_sampler.py:936]: is this for debug only?

Comment (yxy235) [reply tests/python/pytorch/graphbolt/test_subgraph_sampler.py]: Yes. I have deleted it.

**Pull request merged**